### PR TITLE
Enable goss for Azure

### PR DESCRIPTION
**Pull request by EleanorRigby**
Works on VHD and Shared Image Gallery.

SIG
```
==> sig-ubuntu-1804:
Build 'sig-ubuntu-1804' finished.

==> Builds finished. The artifacts of successful builds are:
--> sig-ubuntu-1804: Azure.ResourceManagement.VMImage:
```

VHD
```
==> vhd-ubuntu-1804:
Build 'vhd-ubuntu-1804' finished.

==> Builds finished. The artifacts of successful builds are:
--> vhd-ubuntu-1804: Azure.ResourceManagement.VMImage:
```

GOSS Output
```
Running goss tests...
    vhd-ubuntu-1804: {
    vhd-ubuntu-1804:     "results": [
    vhd-ubuntu-1804:         {
    vhd-ubuntu-1804:             "duration": 14145852,
    vhd-ubuntu-1804:             "err": null,
    vhd-ubuntu-1804:             "expected": [
    vhd-ubuntu-1804:                 "true"
    vhd-ubuntu-1804:             ],
    vhd-ubuntu-1804:             "found": [
    vhd-ubuntu-1804:                 "true"
    vhd-ubuntu-1804:             ],
    vhd-ubuntu-1804:             "human": "",
    vhd-ubuntu-1804:             "meta": null,
    vhd-ubuntu-1804:             "property": "installed",
    vhd-ubuntu-1804:             "resource-id": "cloud-init",
    vhd-ubuntu-1804:             "resource-type": "Package",
    vhd-ubuntu-1804:             "result": 0,
    vhd-ubuntu-1804:             "successful": true,
    vhd-ubuntu-1804:             "summary-line": "Package: cloud-init: installed: matches expectation: [true]",
    vhd-ubuntu-1804:             "test-type": 0,
    vhd-ubuntu-1804:             "title": ""
    vhd-ubuntu-1804:         },
    vhd-ubuntu-1804:         {
    vhd-ubuntu-1804:             "duration": 7955385,
    vhd-ubuntu-1804:             "err": null,
    vhd-ubuntu-1804:             "expected": [
    vhd-ubuntu-1804:                 "true"
    vhd-ubuntu-1804:             ],
    vhd-ubuntu-1804:             "found": [
    vhd-ubuntu-1804:                 "true"
    vhd-ubuntu-1804:             ],
    vhd-ubuntu-1804:             "human": "",
    vhd-ubuntu-1804:             "meta": null,
    vhd-ubuntu-1804:             "property": "enabled",
    vhd-ubuntu-1804:             "resource-id": "containerd",
    vhd-ubuntu-1804:             "resource-type": "Service",
    vhd-ubuntu-1804:             "result": 0,
    vhd-ubuntu-1804:             "successful": true,
    vhd-ubuntu-1804:             "summary-line": "Service: containerd: enabled: matches expectation: [true]",
    vhd-ubuntu-1804:             "test-type": 0,
    vhd-ubuntu-1804:             "title": ""
    vhd-ubuntu-1804:         },
    vhd-ubuntu-1804:         {
    vhd-ubuntu-1804:             "duration": 6012465,
    vhd-ubuntu-1804:             "err": null,
    vhd-ubuntu-1804:             "expected": [
    vhd-ubuntu-1804:                 "true"
    vhd-ubuntu-1804:             ],
    vhd-ubuntu-1804:             "found": [
    vhd-ubuntu-1804:                 "true"
    vhd-ubuntu-1804:             ],
    vhd-ubuntu-1804:             "human": "",
    vhd-ubuntu-1804:             "meta": null,
    vhd-ubuntu-1804:             "property": "running",
    vhd-ubuntu-1804:             "resource-id": "containerd",
    vhd-ubuntu-1804:             "resource-type": "Service",
    vhd-ubuntu-1804:             "result": 0,
    vhd-ubuntu-1804:             "successful": true,
    vhd-ubuntu-1804:             "summary-line": "Service: containerd: running: matches expectation: [true]",
    vhd-ubuntu-1804:             "test-type": 0,
    vhd-ubuntu-1804:             "title": ""
    vhd-ubuntu-1804:         },
    vhd-ubuntu-1804:         {
    vhd-ubuntu-1804:             "duration": 22849245,
    vhd-ubuntu-1804:             "err": null,
    vhd-ubuntu-1804:             "expected": [
    vhd-ubuntu-1804:                 "0"
    vhd-ubuntu-1804:             ],
    vhd-ubuntu-1804:             "found": [
    vhd-ubuntu-1804:                 "0"
    vhd-ubuntu-1804:             ],
    vhd-ubuntu-1804:             "human": "",
    vhd-ubuntu-1804:             "meta": null,
    vhd-ubuntu-1804:             "property": "exit-status",
    vhd-ubuntu-1804:             "resource-id": "crictl ps",
    vhd-ubuntu-1804:             "resource-type": "Command",
    vhd-ubuntu-1804:             "result": 0,
    vhd-ubuntu-1804:             "successful": true,
    vhd-ubuntu-1804:             "summary-line": "Command: crictl ps: exit-status: matches expectation: [0]",
    vhd-ubuntu-1804:             "test-type": 0,
    vhd-ubuntu-1804:             "title": ""
    vhd-ubuntu-1804:         }
    vhd-ubuntu-1804:     ],
    vhd-ubuntu-1804:     "summary": {
    vhd-ubuntu-1804:         "failed-count": 0,
    vhd-ubuntu-1804:         "summary-line": "Count: 4, Failed: 0, Duration: 0.023s",
    vhd-ubuntu-1804:         "test-count": 4,
    vhd-ubuntu-1804:         "total-duration": 23025448
    vhd-ubuntu-1804:     }
    vhd-ubuntu-1804: }
```
```

### CODE REVIEW

**Comment by EleanorRigby** [comment]
FIxes #281
/assign @CecileRobertMichon
cc: @codenrhoden

**Comment by EleanorRigby** [comment]
fixes #281

**Comment by codenrhoden** [comment]
We will also want to add `hack/ensure-goss.sh` to `make deps-azure`.

Definitely want to make sure we heard from @CecileRobertMichon  to make sure we don't break any of the existing pipelines that are setup.

**Comment by CecileRobertMichon** [comment]
Let me run the existing pipelines from this branch and take a look at the PR. Will get back to you later today. Thanks @EleanorRigby!

**Comment by CecileRobertMichon** [comment]
I'm seeing `error initializing provisioner 'goss': Unknown provisioner goss` when running `make build-azure-vhd-ubuntu-1804`

**Comment by EleanorRigby** [comment]
@CecileRobertMichon : You might be missing goss packer provisioner binary in the system.

Can you run `./hack/ensure-goss.sh`?

**Comment by CecileRobertMichon** [comment]
Can confirm goss is not installed on the container image we use for image builds https://github.com/deis/docker-go-dev/blob/master/rootfs/Dockerfile. Once https://github.com/kubernetes-sigs/image-builder/pull/276 merges we might be able to use that instead, but for now we'll need to add goss to our existing image to avoid breaking all our pipelines before this can merge.

/hold

**Comment by EleanorRigby** [comment]
> Can confirm goss is not installed on the container image we use for image builds https://github.com/deis/docker-go-dev/blob/master/rootfs/Dockerfile. Once #276 merges we might be able to use that instead, but for now we'll need to add goss to our existing image to avoid breaking all our pipelines before this can merge.
> 
> /hold

Got it. 

##### My thoughts on Dockerfile vs Runtime 

I think rather than updating Dockerfile which might have a longer turn around for future merges, You can just run `make deps-azure` in your pipeline on the container after I add `ensure-goss` to it. This will remove the dependency of updating Dockerfile with each new packer, goss versions and keep that image smaller to faster to pull. I see that image is used for variety of projects.

**Comment by CecileRobertMichon** [comment]
Makes sense. In that case, let's add it to the pipeline as part of this PR https://github.com/kubernetes-sigs/image-builder/blob/master/images/capi/packer/azure/.pipelines/build-vhd.yaml. You can add a new step for ensure-goss.

**Comment by EleanorRigby** [comment]
I can definitely do that. 

FYI, before all other work, I would like a main GOSS PR to get merged

- Requires merge of #291

**Comment by EleanorRigby** [comment]
/unhold

@CecileRobertMichon : I have added `ensure-goss` as a dep and both `vhd` and `sig` ubuntu passes all 24 GOSS tests.

This PR is ready to be merged.

File: images/capi/Makefile [hunk 1]
```diff
@@ -60,6 +60,8 @@ deps-azure:
 	hack/ensure-packer.sh
 	hack/ensure-jq.sh
 	hack/ensure-azure-cli.sh
+	hack/ensure-goss.sh
```
Comment (CecileRobertMichon) [inline images/capi/Makefile:63]: I'm still seeing `error initializing provisioner 'goss': Unknown provisioner goss` on the pipeline, looks like `make deps-azure` is not getting called. Can you please add it to https://github.com/kubernetes-sigs/image-builder/blob/master/images/capi/packer/azure/.pipelines/build-vhd.yaml#L40?

Comment (EleanorRigby) [reply images/capi/Makefile:63]: @CecileRobertMichon : Sorry, I missed updating azure `.pipeline` folder. Just pushed a commit.

**Comment by CecileRobertMichon** [comment]
lgtm, pipeline is passing now! @EleanorRigby can you please squash the commits

**Comment by EleanorRigby** [comment]
@CecileRobertMichon : All squashed into one commit.

**Pull request merged**